src/lib/controllers/frame-template/renderer/index.ts

```diff
@@ -0,0 +1,31 @@
+import { ConfigMap } from '../../config-store/config-store';
+import { Frame, createFrame } from '../../frame/frame';
+import { FrameTemplate } from '..';
+
+export type FrameTemplateRenderer = (
+	content: string,
+	configMap: ConfigMap,
+) => string;
+
+export interface PreparedFrameTemplateRenderer {
+	render: (content: string) => string;
+	renderTemplate: (template: FrameTemplate) => Frame;
+}
+
+export const prepareRenderer = (
+	render: FrameTemplateRenderer,
+	configMap: ConfigMap,
+): PreparedFrameTemplateRenderer => {
+	return {
+		render: (content: string) => render(content, configMap),
+		renderTemplate: (template: FrameTemplate) => {
+			const frame = createFrame();
+
+			Object.keys(template.files).forEach(path => {
```

Comment:
Assuming `frame.files` doesn't have any existing info you can do this as:
```suggestion
			frame.files = _.mapValues(template.files, file => render(file, configMap)
```